src/security/core/tests/common/handshake_test_utils.c

```diff
@@ -22,6 +22,9 @@
 
 #define TIMEOUT DDS_SECS(2)
 
+static const char * validatation_result_str[] = { "OK", "FAILED", "PENDING_RETRY", "PENDING_HANDSHAKE_REQUEST", "PENDING_HANDSHAKE_MESSAGE", "OK_FINAL_MESSAGE" };
```

Comment:
There's a downside to doing the enum-to-string mapping this way (and there's a typo in the name besides) and that is that there's no compile-time check that all enum symbols are covered correctly. The old switch statement comes a lot closer, although in my view that one has a default case that shouldn't be there. If there's no need for an array, I much rather see a switch, but then reformatted and tweaked a bit, e.g.,
```C
static char * get_validation_result_str(DDS_Security_ValidationResult_t result)
{
  switch (result)
  {
    case DDS_SECURITY_VALIDATION_OK: return "ok";
    case DDS_SECURITY_VALIDATION_PENDING_RETRY: return "pending retry";
    case DDS_SECURITY_VALIDATION_PENDING_HANDSHAKE_REQUEST: return "handshake request";
    case DDS_SECURITY_VALIDATION_PENDING_HANDSHAKE_MESSAGE: return "handshake message";
    case DDS_SECURITY_VALIDATION_OK_FINAL_MESSAGE: return "ok final";
    case DDS_SECURITY_VALIDATION_FAILED: return "failed";
  }
  abort ();
  return "";
}
```
that'll cause a compiler warning if a case has been missed; the risk is that some compilers may warn about the abort/return at the end being unreachable ...

(And if you do need to use an array, designated initializers are nice, like ``[DDS_SECURITY_VALIDATION_OK] = "ok"``, etc.)